mlflow/azureml/__init__.py

```diff
@@ -380,15 +382,41 @@ def deploy(
             _copy_file_or_tree(src=absolute_model_path, dst=model_directory_path),
         )
 
-        registered_model = AzureModel.register(
-            workspace=workspace, model_path=tmp_model_path, model_name=model_name, tags=tags
-        )
+        registered_model = None
+
+        try:
+            if model_uri.startswith("models:/"):
+                m_name = model_uri.split("/")[-2]
+                m_version = int(model_uri.split("/")[-1])
+                m_id = "{}:{}".format(m_name, m_version)
+                registered_model = AzureModel(workspace, id=m_id)
+
+                _logger.info("Found registered model in AzureML with ID '%s'", m_id)
+            elif model_uri.startswith("runs:/") and get_tracking_uri().startswith("azureml"):
+                m = mlflow_register_model(model_uri, model_name)
+                m_id = "{}:{}".format(m.name, m.version)
+                registered_model = AzureModel(workspace, id=m_id)
+
+                _logger.info(
+                    "Registered an Azure Model with name: `%s` and version: `%s`",
+                    registered_model.name,
+                    registered_model.version,
+                )
+        except Exception:  # pylint: disable=broad-except
```

Comment:
Do we need to catch Exception? If we could scope this down it will help against bugs leaking through from unhandled exceptions

Reply:
I think we can pull the if elif out of the try to scope down possible buggy code

then 

if m_name, m_version, m_id are not none,

try:
 register_model
except AzureMLException or even Exception at this point

do the rest